nexus/src/app/sagas/region_replacement_drive.rs

```diff
@@ -0,0 +1,1521 @@
+                        sagactx.lookup::<Uuid>("job_id")?,
+                    ),
+                },
+
+                volume_id: new_region.volume_id(),
+            }
+        } else {
+            // XXX what other volumes are created?
+            return Err(ActionError::action_failed(format!(
+                "don't know what to do with volume {}",
+                new_region.volume_id(),
+            )));
+        }
+    };
+
+    Ok(drive_action)
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ExecuteResult {
+    step_to_commit: Option<db::model::RegionReplacementStep>,
+    replacement_done: bool,
+}
+
+/// Attempt to execute the prepared step. If it was successful, return the step
+/// to commit to the database.
+async fn srrd_drive_region_replacement_execute(
+    sagactx: NexusActionContext,
+) -> Result<ExecuteResult, ActionError> {
+    let log = sagactx.user_data().log();
+    let osagactx = sagactx.user_data();
+    let params = sagactx.saga_params::<Params>()?;
+
+    let opctx = crate::context::op_context_for_saga_action(
+        &sagactx,
+        &params.serialized_authn,
+    );
+
+    let nexus = osagactx.nexus();
+
+    // Look up the prepared action, and execute it. If something has changed
+    // between when the action was determined and now, then bail out - the next
+    // drive saga invocation will pick up the new state of the world and act
+    // accordingly.
+
+    let action = sagactx.lookup::<DriveAction>("prepare")?;
+
+    let result = match action {
+        DriveAction::Noop { replacement_done } => {
+            // *slaps knees and stands up* welp
+            ExecuteResult { step_to_commit: None, replacement_done }
+        }
+
+        DriveAction::Pantry { step, volume_id } => {
+            // Importantly, _do not use `call_pantry_attach_for_disk`_! That
+            // uses `retry_until_known_result`, which we _do not want here_. The
+            // Pantry attach can fail if there's a racing Volume checkout to be
+            // sent to Propolis. Additionally, it uses `attach` instead of
+            // `attach_activate_background`, which means it will hang on the
+            // activation.
+
+            let Some(pantry_address) = step.pantry_address() else {
+                return Err(ActionError::action_failed(String::from(
+                    "pantry step does not have an address",
+                )));
+            };
+
+            let endpoint = format!("http://{}", pantry_address);
+            let client = crucible_pantry_client::Client::new(&endpoint);
+
+            // Check pantry first, to see if this volume is attached already. This can
+            // occur if:
+            //
+            // - the volume is attached to the target pantry, but it can't be reliably
+            //   determined if reconcilation finished.
+            //
+            // - a previous repair operated on another region in the same Volume, and that
```

Comment:
Detaching would cancel it - Nexus would then shortly then attach the Volume and the reconciliation would start up again.

However, in this function, in this match arm, the saga had determined that an action needed to be taken, which means it wasn't aware of this repair, or it determined that repair didn't complete or that it failed. Otherwise it would have chosen `DriveAction::Noop` as a result of `DriveCheck::LastStepStillRunning`.